src/aws-cpp-sdk-core/source/utils/crypto/crt/CRTSymmetricCipher.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/core/utils/crypto/crt/CRTSymmetricCipher.h>
+
+namespace Aws
+{
+    namespace Utils
+    {
+        namespace Crypto
+        {
+            CRTSymmetricCipher::CRTSymmetricCipher(Crt::Crypto::SymmetricCipher &&toMove) : SymmetricCipher(), m_cipher(std::move(toMove))
+            {
+                if (m_cipher)
+                {
+                    auto ivCur = m_cipher.GetIV();
+                    m_initializationVector = CryptoBuffer(ivCur.ptr, ivCur.len);
+
+                    auto keyCur = m_cipher.GetKey();
+                    m_key = CryptoBuffer(keyCur.ptr, keyCur.len);
+
+                    auto tagCur = m_cipher.GetTag();
+
+                    if (tagCur.len)
+                    {
+                        m_tag = CryptoBuffer(tagCur.ptr, tagCur.len);
+                    }
+                }
+            }
+
+            CryptoBuffer CRTSymmetricCipher::EncryptBuffer(const CryptoBuffer &unEncryptedData)
+            {
+                auto resultBuffer = Crt::ByteBufInit(get_aws_allocator(), Crt::Crypto::AES_256_CIPHER_BLOCK_SIZE);
+                Crt::ByteCursor toEncrypt = Crt::ByteCursorFromArray(unEncryptedData.GetUnderlyingData(), unEncryptedData.GetLength());
+
+                if (m_cipher.Encrypt(toEncrypt, resultBuffer))
+                {
+                    return {std::move(resultBuffer)};
+                }
+                Crt::ByteBufDelete(resultBuffer);
```

Comment:
how does error handling work here? we just return empty cryptoBuffer and nothing else?

Reply:
yeah thats all from the original [API](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/utils/crypto/Cipher.h#L96-L119) which heavily relied on the `virtual operator bool() const { return Good(); }` operator to check if the cipher was in a erroronus state or not. which is why i change in this PR to catpure the state of being finished or ready. The error handling here would be to check the ciphers boolean operator before using the return value per the original API.